File: frontends/apps/framework/src/features/sessions/session.ts
import { Form } from "./forms"
import { FrontendJournal } from "./journal"

/**
 * 
 */
export class Session {
    id: string
    locale: string
    form: Form
    journal: FrontendJournal
    state: string

    constructor(
        id: string,
        locale: string,
        state: string,
        form: Form,
    ) {
        this.id = id
        this.locale = locale
        this.state = state
        this.form = form
        this.journal = new FrontendJournal()
    }
}

File: core/framework/src/main/java/com/sap/bfx/session/Session.java
package com.sap.bfx.session;

import com.sap.bfx.p13n.Settings;
import lombok.Data;

import java.util.Collection;
import java.util.HashMap;
import java.util.Locale;
import java.util.Map;

@Data
public final class Session {
    private String id;
    private Locale locale;
    private String displayState;
    private Form form;
    private BackendJournal journal;
    private Collection<Settings> settings;
    private String userName;
    private String taskInstanceId;
    private Map<String, Object> additionalData = new HashMap<>();
}
